app/models/reports/view.rb

```diff
@@ -33,5 +33,12 @@ def self.add_column_description(column_description, column_name)
     def self.materialized?
       raise NotImplementedError
     end
+
+    def self.get_refresh_months
+      current_date = Date.today
+      current_day = current_date.day
+      current_month = current_date.beginning_of_month
+      current_day.odd? ? [current_month, current_month << 1] : [current_month, current_month >> ((current_day / 2) + 1)]
```

Comment:
```ruby
current_month << 1
```

could be

```ruby
current_month.prev_month
```

…and…

```ruby
current_month >> ((current_day / 2) + 1)
```

could be

```ruby
current_month.next_month((current_day / 2) + 1)
```

It's easier to grok what's going on.

Reply:
The `.next_month` bit can be further clarified as 

```ruby
month_offset = current_day / 2 + 1
current_month + month_offset.month
```